src/serializer/Emitter.js

```diff
@@ -343,6 +352,9 @@ export class Emitter {
         result = recurse(val.$Description);
         if (result !== undefined) return result;
       }
+    } else if (val instanceof ArrayValue && ArrayValue.isIntrinsicAndHasWidenedNumericProperty(val)) {
+      result = callbacks.onValueWithIdentifier ? callbacks.onValueWithIdentifier(val) : undefined;
```

Comment:
`WithIdentifier` doesn't really apply here. I propose you introduce a different callback.